pytensor/tensor/math.py

```diff
@@ -402,7 +402,8 @@ def max_and_argmax(a, axis=None, keepdims=False):
     a = as_tensor_variable(a)
     axis = check_and_normalize_axes(a, axis)
     if len(axis) == 0:
-        axis = list(range(a.type.ndim))
+        # axis = list(range(a.type.ndim))
+        axis = None
```

Comment:
This is not correct for higher dimensional `a`. The way to check argmax is doing the correct thing is that `x[argmax(x, axis=axis)` should equal `max(x, axis=axis)`. If you have x be like `np.zeros((5, 3))` you'll see it's wrong. 

This was already wrong before btw. We should open an issue as we don't need to fix in this PR. It's enough if we don't change the bevahior in this PR.

Reply:
You may need `keepdims` for the indexing to work correctly. But the point was mostly than axis=() is not the same as axis=None, except for the scalar case